File: Python/underautomation/universal_robots/ssh/tools/sftp/sftp_file_sytem_information.py
import typing
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Ssh.Tools.Sftp import SftpFileSytemInformation as sftp_file_sytem_information

class SftpFileSytemInformation:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = sftp_file_sytem_information()
		else:
			self._instance = _internal
	@property
	def file_system_block_size(self) -> int:
		return self._instance.FileSystemBlockSize
	@property
	def block_size(self) -> int:
		return self._instance.BlockSize
	@property
	def total_blocks(self) -> int:
		return self._instance.TotalBlocks
	@property
	def free_blocks(self) -> int:
		return self._instance.FreeBlocks
	@property
	def available_blocks(self) -> int:
		return self._instance.AvailableBlocks
	@property
	def total_nodes(self) -> int:
		return self._instance.TotalNodes
	@property
	def free_nodes(self) -> int:
		return self._instance.FreeNodes
	@property
	def available_nodes(self) -> int:
		return self._instance.AvailableNodes
	@property
	def sid(self) -> int:
		return self._instance.Sid
	@property
	def is_read_only(self) -> bool:
		return self._instance.IsReadOnly
	@property
	def supports_set_uid(self) -> bool:
		return self._instance.SupportsSetUid
	@property
	def max_name_lenght(self) -> int:
		return self._instance.MaxNameLenght

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Ssh/Tools/Sftp/SftpFileSytemInformation.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.


namespace Ssh.Tools.Sftp {
	/// <summary>
	/// Contains File system information exposed by statvfs@openssh.com request.
	/// </summary>
	public class SftpFileSytemInformation {

		/// <summary>
		/// Gets the file system block size.
		/// </summary>
		/// <returns>The file system block size.</returns>
		public ulong FileSystemBlockSize { get; }

		/// <summary>
		/// Gets the fundamental file system size of the block.
		/// </summary>
		/// <returns>The fundamental file system block size.</returns>
		public ulong BlockSize { get; }

		/// <summary>
		/// Gets the total blocks.
		/// </summary>
		/// <returns>The total blocks.</returns>
		public ulong TotalBlocks { get; }

		/// <summary>
		/// Gets the free blocks.
		/// </summary>
		/// <returns>The free blocks.</returns>
		public ulong FreeBlocks { get; }

		/// <summary>
		/// Gets the available blocks.
		/// </summary>
		/// <returns>The available blocks.</returns>
		public ulong AvailableBlocks { get; }

		/// <summary>
		/// Gets the total nodes.
		/// </summary>
		/// <returns>The total nodes.</returns>
		public ulong TotalNodes { get; }

		/// <summary>
		/// Gets the free nodes.
		/// </summary>
		/// <returns>The free nodes.</returns>
		public ulong FreeNodes { get; }

		/// <summary>
		/// Gets the available nodes.
		/// </summary>
		/// <returns>The available nodes.</returns>
		public ulong AvailableNodes { get; }

		/// <summary>
		/// Gets the sid.
		/// </summary>
		/// <returns>The sid.</returns>
		public ulong Sid { get; }

		/// <summary>
		/// Gets a value indicating whether this instance is read only.
		/// </summary>
		/// <returns><code>true</code> if this instance is read only; otherwise, <code>false</code>.</returns>
		public bool IsReadOnly { get; }

		/// <summary>
		/// Gets a value indicating whether [supports set uid].
		/// </summary>
		/// <returns><code>true</code> if [supports set uid]; otherwise, <code>false</code>.</returns>
		public bool SupportsSetUid { get; }

		/// <summary>
		/// Gets the max name lenght.
		/// </summary>
		/// <returns>The max name lenght.</returns>
		public ulong MaxNameLenght { get; }
	}
}
